larg4/pluginActions/ParticleListAction_service.cc

```diff
@@ -351,9 +362,17 @@ namespace larg4 {
           // which will put a bogus track id value into the sim::IDE object for
           // the sim::SimChannel if we don't check it.
           if (!fParticleList.KnownParticle(fCurrentTrackID)) fCurrentTrackID = sim::NoParticleId;
+	  fTargetIDMap[trackID] = fCurrentTrackID;
           // clear current particle as we are not stepping this particle and
           // adding trajectory points to it
-	  fdroppedTracksSet.insert(trackID);
+	  if (fdroppedTracksMap.find(this->GetParentage(trackID))==fdroppedTracksMap.end())
+	    {
+	      fdroppedTracksMap[this->GetParentage(trackID)]={trackID};
+	    } else
+	    {
+	      fdroppedTracksMap[this->GetParentage(trackID)].insert(trackID);
+	    }
```

Comment:
This code can be achieved equivalently by just typing:

```c++
fdroppedTracksMap[this->GetParentage(trackID)].insert(trackID);
```

If the `std::set` does not exist for `this-GetParentage(trackID)`, it will be created.  Otherwise, the already-stored `std::set` will be returned by calling `std::map::operator[]`.

Reply:
done